### ISSUE TITLE
Provide DOIs for taxon publications

### ISSUE THREAD

**Issue by charvolant** [AtlasOfLivingAustralia/data-management#501]
We have reference information from our sources. Provide DOI links to the references.

* Derived from Rod Page, https://doi.org/10.7717/peerj.6739
* The usual, how do we ensure accuracy?
  * Should we be getting these from NSL/AFD? They can curate the DOIs of the publications.
* There is a ready-made pair of fields for DOIs  nameAccordingToID and namePublishedInID

**Comment by elywallis** [comment]
My recommendation for the record: 
Yes, DOIs for references *should* be provided by NSL / AFD but they are not currently in a position to do this. 
We should go ahead and implement the links as per Rod Page's demonstration because they will be extremely beneficial to *users*
Once NSL / AFD have ingested the DOIs and can provide them, we make any changes to pull the data from there instead of using Rod's solution. 
In short, the benefit to users is great enough that we should simply go ahead now as it may take several years for AFD to sort this out. I don't think we should deny users this opportunity simply because it "ought" to be done diffferently.

**Comment by nielsklazenga** [comment]
I think it would be good to talk with NSL about this. There is a DOI field (as well as ISBN and ISSN fields) in the NSL data model and it seems (to me) a rather trivial change to get it into the editor as well. Pretty sure people who use the NSL services (web pages) directly will want the DOIs as well. This might be just the push the NSL needs to make it a priority. Rod is meeting with NSL in just under an hour, so that might already do it.

**Comment by elywallis** [comment]
Note that the demo Rod has done is only for content in AFD. Yes, it's true that APNI/APC has fields for DOI but my understanding is that Rod has not done the same process he did with AFD to find all the references/DOI content. We need to be clear that *right now* we could only implement the links provided by Rod for AFD.
In addition we could start to bring the DOIs through into ALA from APNI/APC where the content has been populated. That would be a second part of the project.
I think we should stop referring to "NSL" as it's confusing in this context.

**Comment by nielsklazenga** [comment]
I agree that "NSL" could be confusing to some people and we agreed to replace it with something better, but as long as that hasn't happened, it is the only thing we've got and APNI/APC is apparently just as confusing. APNI/APC is vascular plants only. Bryophytes, fungi – incl. lichens – and algae are also in the NSL infrastructure, but not in APNI/APC. Also, while the Ozymandias app. might currently only contain AFD, Rod came to see me a few weeks ago because he wanted to include a plant data set as well. IPNI (International Plant Name Index), which has all the vascular plant names, he has already done (or was doing). So AFD is definitely not the end of it. 

Also, AFD is managed by the NSL folks at the ANBG (the "NSL" you can actually talk to), so they can take care of AFD names and DOIs as well. It's better to keep them as close to the source as possible. 

I think also the DOIs are just for the original publications of the names. These serve a purely administrative purpose and the content of these publications is of very little interest to the majority of taxonomists, let alone most ALA users. There is absolutely no hurry getting these DOIs into ALA and definitely not worth the friction taking action without consultation is going to cause between ALA and NSL.

This is NSL territory, not ALA data management.

**Comment by charvolant** [comment]
An additional idea from @nickdos We need a general administrative interface that allows us to override wrong information (images, links to EoL, BHL, DOIs etc. etc.) without needing to re-code things. Basically bite the bullet, accept that there will always be the 1% and not try to deal with it algorithmically. Similar to the preferred image process.

This would require some re-thinking of how the bie-index works. We'll also need to figure out how to accommodate migration of over-rides when things change.
